### PR TITLE
don't fail during preprocess for sampling from iterable dataset

### PR DESCRIPTION
**Pull request by winglian**
## Summary by CodeRabbit

- **Bug Fixes**
  - Improved handling of iterable datasets to prevent errors during debug checks, ensuring smoother operation when sampling and checking labels.

### CODE REVIEW

File: src/axolotl/common/datasets.py [hunk 1]
```diff
@@ -75,13 +75,17 @@ def load_datasets(
...
+        try:
+            train_samples = sample_dataset(train_dataset, num_examples)
+            check_dataset_labels(
+                train_samples,
+                tokenizer,
+                num_examples=num_examples,
+                text_only=text_only,
+            )
+        except AttributeError:
+            # can't sample iterable datasets
```
Comment (salmanmohammadi) [inline src/axolotl/common/datasets.py:87]: Let's add a log to indicate this to the user.

**Review by salmanmohammadi — APPROVED** [review]
